pkg/pulumiyaml/sort.go

```diff
@@ -73,6 +104,58 @@ func topologicallySortedResources(t *ast.TemplateDecl) ([]graphNode, syntax.Diag
 			sorted = append(sorted, node)
 		}
 	}
+	var projConf map[string]interface{}
+	var projConfStr = os.Getenv("PULUMI_CONFIG")
+	if err := json.Unmarshal([]byte(projConfStr), &projConf); err == nil {
```

Comment:
If there is an error unmarshalling, we need to report it. We should almost never swallow an error.